real_decoder.go

```diff
@@ -143,11 +155,10 @@ func (rd *realDecoder) getString() (string, error) {
 }
 
 func (rd *realDecoder) getNullableString() (*string, error) {
-	tmp, err := rd.getInt16()
-	if err != nil || tmp == -1 {
+	str, err := rd.getString()
+	if err != nil || str == "" {
```

Comment:
I think the whole point of nullable strings was to distinguish between a length of `-1` (null) and a length of `0` (`""`) which this check doesn't do, because `getString` already turns them both into `""`.

Reply:
I factored out `getStringLength()` now to allow for exactly this logic. the problem with the former `getNullableString()` impl was that it consumed the length so when calling out to `getString()` the output became corrupted.